cmd/client.go

```diff
@@ -171,16 +175,10 @@ func (cpc *CloudproberClient) RemoveProbe(probeName string) error {
 // - ListProbesResponse: This is the list of active probes in Cloudprober.
 //			 If this is empty, there are no active probes.
 // - Error:
-//	   - ClientNotInitialised: the gRPC client of CloudproberClient is not registered.
-//	     -> Solution: Call the InitClient(rpcServer string) method.
-func (cpc *CloudproberClient) ListProbes() (*proberpb.ListProbesResponse, error) {
-	cpc.clientMux.Lock()
-	defer cpc.clientMux.Unlock()
-
-	if cpc.client == nil {
-		err := fmt.Errorf("ClientNotInitialised: cannot register and add a probe when client is not initialised")
-		return nil, err
-	}
+//	   - See Cloudprober ListProbes() RPC for details on an error.
+func (client *CloudproberClient) ListProbes() (*proberpb.ListProbesResponse, error) {
+	client.clientMux.Lock()
```

Comment:
Do we need to lock to list probes?

Reply:
The reason I'm locking is so that if two functions call one of CloudproberClient's methods, e.g. ListProbes() and RemoveProbe(), one is completed after the other due to mutex locking. Is that not necessary here?

Reply:
Does `ListProbes()` do something reasonable if we call it while removeprobes is running, is I suppose my real question?

Reply:
That's a good question. From the Cloudprober code, it seems that ListProbes() does some locking within Cloudprober, so we should be safe.